### ISSUE TITLE
Mesh Shader

### ISSUE THREAD

**Issue by ghost** [xCollateral/VulkanMod#257]
https://github.com/MCRcortex/nvidium

This module uses the `GL_NV_mesh_shader` to optimize rendering performance, and is only available for NVIDIA.

In Vulkan, there is a cross-vendor mesh shader extension: `VK_EXT_mesh_shader`

Use this extension to implement mesh shader rendering in VulkanMod, which should improve performance on modern GPUs (NVIDIA Turing+, Intel DG2+, AMD RDNA2+)

**Comment by Haxorzz** [comment]
Wow, that's an incredible mod. Didn't even know about it. If this mod could make it work on recent AMD gpus it would be amazing! [Here's the blog post for VK_EXT_mesh_shader](https://www.khronos.org/blog/mesh-shading-for-vulkan) it seems to be very similar to nvidia's original turing mesh shading which is what nvidium uses (they also use the same graphs and images to explain it as nvidia). Hopefulyy this is considered!

**Comment by Naitrate** [comment]
Yeah, decided to further look into this mod after seeing a YouTuber cover Nvidium, and it would be really neat to see a more vendor agnostic version of this implemented with this Vulkan renderer.

**Comment by ikumi90s** [comment]
as an Intel ARK user, this would excite me to use since Vulkan with Intel GPU performs better than OPL and DX12

**Comment by Haxorzz** [comment]
According to [this comment](https://github.com/xCollateral/VulkanMod/issues/362#issuecomment-1926706428) it should be possible to get most of the performance gain without mesh shaders, and on basically every card that supports compute shaders!
